presentation/opentelemetry/src/presentation-opentelemetry/Diagnostics.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+/** @packageDocumentation
+ * @module Core
+ */
+import { Diagnostics, DiagnosticsLogEntry, DiagnosticsScopeLogs } from "@itwin/presentation-common";
+import { randomBytes } from "crypto";
+import { HrTime, SpanContext, SpanKind, SpanStatusCode, TraceFlags } from "@opentelemetry/api";
+import { ReadableSpan, TimedEvent } from "@opentelemetry/sdk-trace-base";
+import { Resource } from "@opentelemetry/resources";
+import { SemanticResourceAttributes } from "@opentelemetry/semantic-conventions";
+
+/** @public */
+export function convertToReadableSpans(diagnostics: Diagnostics, parentSpanContext?: SpanContext): ReadableSpan[] {
+  let spans: ReadableSpan[] = [];
+  for (const logs of diagnostics.logs ?? []) {
+    const nestedSpans = convertScopeToReadableSpans(logs, parentSpanContext ? parentSpanContext.traceId : generateTraceId(), parentSpanContext?.spanId);
+    spans = spans.concat(nestedSpans);
+  }
+  return spans;
+}
+
+function convertScopeToReadableSpans(logs: DiagnosticsScopeLogs, traceId: string, parentSpanId?: string): ReadableSpan[] {
+  if (!logs.scopeCreateTimestamp || !logs.duration)
+    return [];
+
+  const spanId = generateSpanId();
+  let spans: ReadableSpan[] = [];
+  const events: TimedEvent[] = [];
+
+  for (const entry of logs.logs ?? []) {
+    if (DiagnosticsLogEntry.isMessage(entry)) {
+      const event: TimedEvent = {
+        time: millisToHrTime(entry.timestamp),
+        name: entry.message,
+        attributes: {
+          category: entry.category,
+          ...(entry.severity.dev ? { devSeverity: entry.severity.dev } : undefined),
+          ...(entry.severity.editor ? { editorSeverity: entry.severity.editor } : undefined),
+        },
+      };
+      events.push(event);
+    } else
+      spans = spans.concat(convertScopeToReadableSpans(entry, traceId, spanId));
```

Comment:
if `if` has curly braces, please also add them to `else`, even if it's just one line